db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/FromBaseTable.java

```diff
@@ -196,36 +196,30 @@
     private AggregateNode aggrForSpecialMaxScan;
 
     private boolean isBulkDelete = false;
+
+    private ValueNode pastTxIdExpression = null;
+
     @Override
     public boolean isParallelizable(){
         return false;
     }
 
     /**
-     * Initializer for a table in a FROM list. Parameters are as follows:
-     * <p/>
-     * <ul>
-     * <li>tableName            The name of the table</li>
-     * <li>correlationName    The correlation name</li>
-     * <li>derivedRCL        The derived column list</li>
-     * <li>tableProperties    The Properties list associated with the table.</li>
-     * </ul>
-     * <p/>
-     * <p>
-     * - OR -
-     * </p>
-     * <p/>
-     * <ul>
-     * <li>tableName            The name of the table</li>
-     * <li>correlationName    The correlation name</li>
-     * <li>updateOrDelete    Table is being updated/deleted from. </li>
-     * <li>derivedRCL        The derived column list</li>
-     * </ul>
+     * Initializer for a table in a FROM list.
+     * @param tableName The name of the table
+     * @param correlationName The correlation name
+     * @param rclOrUD update/delete flag or result column list
+     * @param propsOrRcl properties or result column list
+     * @param isBulkDeleteOrTxId bulk delete flag or past tx id expression.
      */
     @Override
-    public void init(Object tableName,Object correlationName,Object rclOrUD,Object propsOrRcl, Object isBulkDelete){
+    public void init(Object tableName,Object correlationName,Object rclOrUD,Object propsOrRcl, Object isBulkDeleteOrTxId){
+        if(isBulkDeleteOrTxId instanceof Boolean) {
```

Comment:
I'm not a fan of the `isBulkDeleteOrTxId` parameter, would it be possible to add an extra parameter for the transaction id ?

Reply:
Right, I did it in the base change (DB-9714). The new signature looks like this:
`public void init(Object tableName,Object correlationName,Object rclOrUD,Object propsOrRcl, Object isBulkDelete, Object pastTxIdExpr)`